core/state/history.go

```diff
@@ -0,0 +1,101 @@
+package state
+
+import (
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie2"
+	"github.com/NethermindEth/juno/db"
+)
+
+var _ StateReader = &StateHistory{}
+
+type StateHistory struct {
+	blockNum uint64
+	state    *State
+}
```

Comment:
Everything has been amazingly explained so far. Can we add a bit to this struct as well